src/test/software/amazon/qldb/QldbDriverImplTest.java

```diff
@@ -123,6 +124,29 @@ public void testBuildWitNullIonSystem() {
                                      .build());
     }
 
+    @Test
+    public void testBuildWithNullHttpClientBuilder() {
+        assertThrows(NullPointerException.class,
+                     () -> QldbDriver.builder()
+                                     .sessionClientBuilder(mockBuilder)
+                                     .ledger(LEDGER)
+                                     .httpClientBuilder(null)
+                                     .build());
+    }
+
+    @Test
+    public void testBuildWithCustomizedHttpClientBuilder() {
```

Comment:
This test is what I call an "MD5-test". You're writing test code which explicitly states what the production code should do at a syntactic level. Rather verify that, once built, the actual HTTP client in use is the one we want, and not the one the other code path creates. 

That might be really hard to do. In that case, you do not need to write a test. The cost of the test outweighs the value of the test.

Reply:
Deleted test